api/org/commcare/api/json/PromptToJson.java

```diff
@@ -0,0 +1,227 @@
+                break;
+        }
+    }
+
+    private static void parseRepeatJuncture(FormEntryModel model, JSONObject obj, FormIndex ix) {
+        FormEntryCaption formEntryCaption = model.getCaptionPrompt(ix);
+        FormEntryCaption.RepeatOptions repeatOptions = formEntryCaption.getRepeatOptions();
+        parseCaption(formEntryCaption, obj);
+        obj.put("header", repeatOptions.header);
+        obj.put("repetitions", formEntryCaption.getRepetitionsText());
+        obj.put("add-choice", repeatOptions.add);
+        obj.put("delete-choice", repeatOptions.delete);
+        obj.put("del-header", repeatOptions.delete_header);
+        obj.put("done-choice", repeatOptions.done);
+    }
+
+    private static void parseQuestionAnswer(JSONObject obj, FormEntryPrompt prompt){
+        IAnswerData answerValue = prompt.getAnswerValue();
+        if (answerValue == null){
+            obj.put("answer", JSONObject.NULL);
+            return;
+        }
+        switch(prompt.getDataType()) {
+            case Constants.DATATYPE_NULL:
+            case Constants.DATATYPE_TEXT:
+                obj.put("answer", answerValue.getDisplayText());
+                return;
+            case Constants.DATATYPE_INTEGER:
+                obj.put("answer", (int)answerValue.getValue());
+                return;
+            case Constants.DATATYPE_LONG:
+            case Constants.DATATYPE_DECIMAL:
+                obj.put("answer", (double)answerValue.getValue());
+                return;
+            case Constants.DATATYPE_DATE:
+                obj.put("answer", (DateUtils.formatDate((Date)answerValue.getValue(), DateUtils.FORMAT_ISO8601)));
+                return;
+            case Constants.DATATYPE_TIME:
+                obj.put("answer", answerValue.getDisplayText());
+                return;
+            case Constants.DATATYPE_DATE_TIME:
+                obj.put("answer", ((Date)answerValue.getValue()).getTime()/1000);
+                return;
+            case Constants.DATATYPE_CHOICE:
+                Selection singleSelection = ((Selection)answerValue.getValue());
+                singleSelection.attachChoice(prompt.getQuestion());
+                obj.put("answer", ((Selection)answerValue.getValue()).index + 1);
+                return;
+            case Constants.DATATYPE_CHOICE_LIST:
+                Vector<Selection> selections = ((SelectMultiData)answerValue).getValue();
+                JSONArray acc = new JSONArray();
+                for(Selection selection: selections){
+                    selection.attachChoice(prompt.getQuestion());
+                    int ordinal = selection.index + 1;
+                    acc.put(ordinal);
+                }
+                obj.put("answer", acc);
+                return;
+            case Constants.DATATYPE_GEOPOINT:
+                double[] coords = ((GeoPointData)prompt.getAnswerValue()).getValue();
+                obj.put("answer", Arrays.copyOfRange(coords, 0, 2));
```

Comment:
Might be more robust to access this data via `GeoPointData.getLatitude()` and `GeoPointData.getLongitude()`, given that if the internal representation changes we don't need to update this code.